### PR TITLE
feat: update dependencies and refactor project dependencies management

### PR DESCRIPTION
**Pull request by reznord**
### Description

As I have mentioned in https://github.com/1Hive/gardens-ui/issues/785, the current 1hive-ui library ships a bunch of libraries which are really not necessary to be shipped along with the component library.

This is an attempt in reducing the foot-print of the library by moving the dependencies to peerDependencies.

In this PR, the following things needs to be done:

- [x] Update all the dependencies of 1hive-ui library
- [ ] Add Changelog to the repo with the decisions

### Dependency Update checklist

Here I will be maintaining a separate checklist for the dependencies which are updated and which yet needs to be updated.

- [x] Babel
- [x] ESLint
- [x] React-spring
- [x] React & React DOM
- [x] Styled Components
- [x] Rollup & its dependencies

* Should I update these dependencies in the same PR or should I update them in a separate one?

### Library size metrics:

I was able to extract the bundle size metrics by running the `yarn build:analyze` script and these are the following results before and after the changes.

#### Before:
image

#### After:
image

### Note: 
This can be just a minor update to the library since we are just updating all the dependencies of the project.

### CODE REVIEW

**Comment by 0xGabi** [comment]
@reznord is this one ready to review?


>Should I update these dependencies in the same PR or should I update them in a separate one?

What do you think of deciding depending on the size of the PR and how extra work would be for you to split them?

**Comment by reznord** [comment]
@0xGabi once I am done with the changes on the [`react-spring` library update PR](#32), then we can merge this one and release a new major version :) 

> What do you think of deciding depending on the size of the PR and how extra work would be for you to split them?

I am already doing that in a separate PR (using this as the base branch), so it will be one single PR to the end to merge.

File: src/components/Popover/Popover.js [hunk 1]
```diff
@@ -45,6 +45,7 @@ class PopoverBase extends React.Component {
   _popper = null
 
   componentDidMount() {
+    console.log(this.props.opener)
```
Comment (kanaagawa) [inline src/components/Popover/Popover.js]: Let's remember to remove this log.

**Review by 0xGabi — APPROVED** [review]
One step closer to the next.js migration 🙌

Amazing work! 🔥🚒

File: src/components/Popover/Popover.js (hunk 1 above)
Comment (reznord) [reply src/components/Popover/Popover.js]: ah yes, forgot that. Will remove it :)

File: devbox/apps/AppBarTabs.js [hunk 2]
```diff
@@ -38,6 +37,8 @@ const App = () => {
 
   const items = tab === null ? ITEMS : ITEMS.slice(0, 1)
 
+  console.log('tab', tab)
```
Comment (kamikazebr) [inline devbox/apps/AppBarTabs.js]:
```suggestion
```

**Review by kamikazebr — CHANGES_REQUESTED** [review]
Awesome!
I left a suggestion.

**Review by 0xGabi — APPROVED** [review]

**Pull request merged**